### PR TITLE
[ONNX] Add support for RandomUniform operator

### PR DESCRIPTION
**Pull request by mateusztabaka**
Ticket: 49689

### CODE REVIEW

File: ngraph/frontend/onnx_import/src/op/random_uniform.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                                   [](int64_t dim) -> size_t { return static_cast<size_t>(dim); });
+
+                    using ONNX_NAMESPACE::TensorProto;
+                    TensorProto::DataType dtype =
+                        static_cast<TensorProto::DataType>(node.get_attribute_value<int64_t>(
+                            "dtype", static_cast<int64_t>(TensorProto::FLOAT)));
+                    switch (dtype)
+                    {
+                    case TensorProto::FLOAT:
+                        return detail::gen_random_constant<float>(shape, low, high, seed);
```
Comment (tsocha) [inline ngraph/frontend/onnx_import/src/op/random_uniform.cpp:76]: Is returning a constant a correct behavior here?
It have sense only if seed is pre-definied in attributes IMO.

**Review by tsocha — CHANGES_REQUESTED** [review]

File: ngraph/frontend/onnx_import/src/op/random_uniform.cpp (hunk 1 above)
Comment (mateusztabaka) [reply ngraph/frontend/onnx_import/src/op/random_uniform.cpp:72]: Can you elaborate more? Why is it not okay to return constant if seed attribute is not provided and we have to generate one?

Comment (tsocha) [reply ngraph/frontend/onnx_import/src/op/random_uniform.cpp:72]: On a second thought I think that even if seed is pre-definied we should not return a constant value.
This approach works only if a model is run only once after import.
But if the model is imported and then several iterations are run we won't have random values between iterations.
Current solution is similar to roll a dice and hardcode a score in code.
I don't think it is a proper way to implement this operator.

Comment (mateusztabaka) [reply ngraph/frontend/onnx_import/src/op/random_uniform.cpp:72]: Yes, you're right.

**Pull request closed without merging**